src/server.ts

```diff
@@ -25,8 +27,39 @@ const authenticationBasePath = container.resolve<string>(
 const APIBasePath = container.resolve<string>("APIBasePath");
 const PagoPABasePath = container.resolve<string>("PagoPABasePath");
 
+// Set default for graceful-shutdown
+const DEFAULT_SHUTDOWN_SIGNALS = "SIGINT SIGTERM";
+const DEFAULT_SHUTDOWN_TIMEOUT_MILLIS = 30000;
+
+const shutdownSignals: string =
+  process.env.SHUTDOWN_SIGNALS || DEFAULT_SHUTDOWN_SIGNALS;
+
+const shutdownTimeout: number = process.env.DEFAULT_SHUTDOWN_TIMEOUT_MILLIS
+  ? parseInt(process.env.DEFAULT_SHUTDOWN_TIMEOUT_MILLIS, 10)
+  : DEFAULT_SHUTDOWN_TIMEOUT_MILLIS;
+
 // tslint:disable-next-line: no-let
 let server: http.Server | https.Server;
+
+export const serverShutdown = (app: Express) => {
+  function cleanup(): Promise<void> {
+    return new Promise(resolve => {
+      log.info("Gracefully shutting down in progress");
+      server.close(() => {
+        app.emit("server:stop");
```

Comment:
this event gets already emitted by [this handler](https://github.com/teamdigitale/io-backend/pull/431/files#diff-916156c481141bfcd08c6ae40ce10e00R88) on `server.close`

Reply:
Ok I remove the cleanup functions, the server close is already emitted. Regarding this question: "Moreover, you aren't testing anything here, how are you making sure the server waits for the active connections to complete before shutting down?" The response is: the server.close() method stops the HTTP server from accepting new connections, and keeps existing connections.

Reply:
Removed in e460176